### PR TITLE
feat(vrl): add `seahash` function

### PR DESCRIPTION
**Pull request by psemeniuk**
Resolves [16072](https://github.com/vectordotdev/vector/issues/16072)

Rationale in the linked issue. 

Note that function converts unsigned int-64 seahash to signed int-64 to met VRL standards, so integer overflow happens when calculated hash is higher than signed int-64 max value. Depending on the use-case this might or mightn't be an issue.

### CODE REVIEW

File: website/cue/reference/remap/functions/sha1.cue [hunk 1]
```diff
@@ -15,7 +15,7 @@ remap: functions: sha1: {
 		},
 	]
 	internal_failure_reasons: []
-	return: types: ["string"]
+	return: types: ["integer"]
```
Comment (bruceg) [inline website/cue/reference/remap/functions/sha1.cue:18]: I'm not clear why this was changed. I don't see the return type of the `sha1` function being changed?

File: website/cue/reference/remap/functions/seahash.cue [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	arguments: [
+		{
+			name:        "value"
+			description: "The string to calculate the hash for."
+			required:    true
+			type: ["string"]
+		},
+	]
+	internal_failure_reasons: []
+	return: types: ["string"]
```
Comment (bruceg) [inline website/cue/reference/remap/functions/seahash.cue:19]: I think this should be where that `integer` from the `sha1` docs belongs.

File: website/cue/reference/remap/functions/seahash.cue [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package metadata
+
+remap: functions: seahash: {
+	category:    "Cryptography"
+	description: """
+		Calculates a [Seahash](\(urls.seahash)) hash of the `value`.
+		Note: function converts unsigned int-64 seahash to signed int-64 to met vrl standards, so integer overflow happens when calculated hash is higher than signed int-64 max value. Depending on the use-case this might or mightn't be an issue.
```
Comment (bruceg) [inline website/cue/reference/remap/functions/seahash.cue:7]:
```suggestion
		Note: Due to limitations in the underlying VRL data types, this function converts the unsigned 64-bit integer seahash result to a signed 64-bit integer. Results higher than the signed 64-bit integer maximum value wrap around to negative values.
```

**Review by bruceg — CHANGES_REQUESTED** [review]
I have a few questions about the documentation, but the code looks fine. Thanks for the contribution.

File: website/cue/reference/remap/functions/seahash.cue (hunk 2 above)
Comment (psemeniuk) [reply website/cue/reference/remap/functions/seahash.cue]: Yes exactly. Thanks for noticing.

**Review by bruceg — APPROVED** [review]

**Comment by psemeniuk** [comment]
I have to admin that previously i was not even run the tests, because i had problems with environment setup - sorry for that. 
This time i built, ran tests, and checked in the VRL CLI, so everything should be fine.

**Review by bruceg — APPROVED** [review]

**Comment by psemeniuk** [comment]
@bruceg I think everything is ready. Can we go further? I don't want to miss the next release :)

**Review by jszwedko — APPROVED** [review]
Thanks for this contribution @psemeniuk !

It's a bit unfortunate that we don't have a `u64` value type to store the result, but that doesn't seem like enough to block this.

**Comment by jszwedko** [comment]
@psemeniuk looks like there are some clippy errors when you get a moment.

**Comment by psemeniuk** [comment]
Ah, you have a bulletproof pipeline :) 
@jszwedko Could you trigger quality gates once again? I hope this was the last one.

**Comment by spencergilbert** [comment]
> Btw - can you share some ETA for the next release?

We're hoping the `0.28.0` release comes out on February 28th!

**Pull request merged**